pymdownx/magiclink.py

```diff
@@ -84,10 +94,24 @@ def handleMatch(self, m):
 class MagiclinkExtension(Extension):
     """Add Easylink extension to Markdown class."""
 
+    def __init__(self, *args, **kwargs):
+        """Initialize."""
+
+        self.config = {
+            'strip_protocol': [
+                False,
+                "If 'True', links are displayed without the initial ftp://, http:// or https://"
+                "- Default: False"
+            ]
+        }
+        super(MagiclinkExtension, self).__init__(*args, **kwargs)
+
     def extendMarkdown(self, md, md_globals):
         """Add support for turning html links and emails to link tags."""
 
-        md.inlinePatterns.add("magic-link", MagiclinkPattern(RE_LINK, md), "<not_strong")
+        md.inlinePatterns.add("magic-link",
+                              MagiclinkPattern(RE_LINK, md, strip_protocol=self.getConfigs()['strip_protocol']),
```

Comment:
I would prefer if you attached the config to the class as the other extensions do: https://github.com/facelessuser/pymdown-extensions/blob/master/pymdownx/progressbar.py#L251.